godtossing/Pos/Ssc/GodTossing/Core/Types.hs

```diff
@@ -37,42 +38,50 @@ module Pos.Ssc.GodTossing.Core.Types
        , NodeSet
        ) where
 
+import           Control.Lens        (each, traverseOf)
 import           Data.Hashable       (Hashable (..))
+import           Data.HashMap.Strict (HashMap)
 import qualified Data.HashMap.Strict as HM
 import qualified Data.Text.Buildable
 import           Formatting          (bprint, build, int, (%))
 import           Universum
 
-import           Pos.Binary.Class    (AsBinary (..))
+import           Pos.Binary.Class    (AsBinary (..), fromBinaryM, serialize')
 import           Pos.Binary.Core     ()
 import           Pos.Core.Address    (addressHash)
 import           Pos.Core.Types      (EpochIndex, StakeholderId)
-import           Pos.Crypto          (EncShare, Hash, PublicKey, Secret, SecretKey,
-                                      SecretProof, SecretSharingExtra, Share,
-                                      SignTag (SignVssCert), Signature, VssPublicKey,
-                                      checkSig, sign, toPublic)
+import           Pos.Crypto          (DecShare, EncShare, Hash, PublicKey, Secret,
+                                      SecretKey, SecretProof, SignTag (SignVssCert),
+                                      Signature, VssPublicKey, checkSig, sign, toPublic)
 
 type NodeSet = HashSet StakeholderId
 
 ----------------------------------------------------------------------------
 -- Commitments
 ----------------------------------------------------------------------------
 
--- | Commitment is a message generated during the first stage of
--- GodTossing. It contains encrypted shares and proof of secret.
--- Invariant which must be ensured: commShares is not empty.
+-- | Commitment is a message generated during the first stage of GodTossing.
+-- It contains encrypted shares and proof of secret.
+--
+-- There can be more than one share generated for a single participant.
 data Commitment = Commitment
-    { commExtra  :: !(AsBinary SecretSharingExtra)
-    , commProof  :: !(AsBinary SecretProof)
-    , commShares :: !(HashMap (AsBinary VssPublicKey) (NonEmpty (AsBinary EncShare)))
+    { commProof  :: !SecretProof
```

Comment:
https://github.com/input-output-hk/cardano-sl/blob/db17c31720f7ec885576c6bc33b9c279d61f1450/core/Pos/Binary/Crypto.hs#L104
There is my note which you forgot to update.